src/TableV8/styled.tsx

```diff
@@ -116,25 +153,108 @@ const TableRow = (
     withStripes?: boolean;
   }>,
 ) => <tr {...props} className={clsx(TABLE_CLASSES.TR, props.className)} />;
-export const StyledTableRow = styled(TableRow, {
-  shouldForwardProp: (prop) =>
-    isPropValid(prop) &&
-    !['index', 'withRowBorder', 'withRowHighlight', 'withStripes'].includes(prop),
+export const TableRow = styled(TableRowComp, {
+  shouldForwardProp: (prop) => isPropValid(prop),
 })`
-  display: flex;
   background: ${(props) =>
-    props.index % 2 === 0 && props.withStripes ? COLORS.BACKGROUND : 'none'};
-  border-bottom: ${(props) => `1px solid ${props.withRowBorder ? COLORS.BORDER : 'transparent'}`};
-  &:hover {
-    background: ${(props) =>
-      props.withRowHighlight ? COLORS.ROW_HIGHLIGHT_BACKGROUND : 'initial'};
-  }
+    props.index % 2 && props.withStripes ? COLORS.BACKGROUND_ALT : 'transparent'};
+  border-bottom: ${(props) => (props.withRowBorder ? `1px solid ${COLORS.BORDER}` : '0 none')};
+  ${(props) =>
+    props.withRowHighlight &&
+    `
+    &:hover {
+      background: ${COLORS.BACKGROUND_HIGHLIGHT}
+    }
+  `}
 `;
 
-const TableCell = (props: React.PropsWithChildren<{ className?: string }>) => (
+// the TD element should only for table structure.
+// use TableCellWrapper for additional styling within the TD.
```

Comment:
there looks like a couple of cases where you intend to limit the consumer of the component eg. you don't want them to style TableCell, you want them to do any styling with TableCellWrapper. I don't think comments in the source are necessarily the best way to enforce this. What are your thoughts on not passing a `className` prop, therefore limiting `styled`, for those components? 

(ofc they can always just throw in inline styles but we have bigger quality issues if that's the case)

Reply:
good points. I'm not a fan of this initial approach because I want to enforce using TableCellWrapper plus adding the meta object and I'm sure people will forget to do one or the other. once I finish adding the tables I'm going to limit which components get access to styled/classnames. also I'm thinking about making a customized columns format that would then get processed in uikit into React Table's column format.

e.g. instead of `{ cell: <TableCellWrapper />, meta: { customCell: true } }` change it to `{ customCell: <TableCellWrapper /> }`.